### PR TITLE
Wrap kubetoken client into OktetoClient

### PR DESCRIPTION
**Pull request by teresaromero**
# Proposed changes

this PR wraps the existent kubetoken client code into the OktetoClient interface in order to be able to create testeable flows with this client.

Also, added new function CheckService to request a HEAD call to check if the service is available at the cluster

### CODE REVIEW

**Review by andreafalzetti — APPROVED** [review]

File: pkg/types/kubetoken.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"encoding/json"
+	authenticationv1 "k8s.io/api/authentication/v1"
+)
+
+type KubeTokenResponse struct {
+	authenticationv1.TokenRequest
+}
+
+func (k *KubeTokenResponse) ToJson() (string, error) {
```
Comment (jLopezbarb) [inline pkg/types/kubetoken.go:12]: NIT: types package should not contain any method. ToJson should be a struct that transform a kubetoken to json. If we want to extend functionality and create a `ToYAML`  we don't have to modify `KubeTokenResponse` WDYT?

File: pkg/types/kubetoken.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package types
+
+import (
+	"encoding/json"
+	authenticationv1 "k8s.io/api/authentication/v1"
+)
+
+type KubeTokenResponse struct {
```
Comment (jLopezbarb) [inline pkg/types/kubetoken.go:8]: Should we add a comment here?

File: pkg/okteto/kubetoken_client.go [hunk 3]
```diff
@@ -14,56 +14,92 @@
...
+	var kubeTokenResponse types.KubeTokenResponse
+	err = json.Unmarshal(body, &kubeTokenResponse)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal kubetoken response: %w", err)
+	}
+
+	return &kubeTokenResponse, nil
+}
+
+func (c *kubeTokenClient) CheckService(baseURL, namespace string) error {
```
Comment (jLopezbarb) [inline pkg/okteto/kubetoken_client.go:85]: Just to make sure. This function is only used on tests.Should we call it when we init the kubetoken client?

**Review by jLopezbarb — APPROVED** [review]
Minor comments only. LGTM!!

File: pkg/okteto/kubetoken_client.go (hunk 3 above)
Comment (teresaromero) [reply pkg/okteto/kubetoken_client.go:85]: this is going to be used in the following PRs, we added as both andrea and me will needed for our issues

File: pkg/types/kubetoken.go (hunk 1 above)
Comment (andreafalzetti) [reply pkg/types/kubetoken.go]: makes sense, done in 1945273c3da13950ac31a1fe3162716b6e580d2e

**Review by teresaromero — COMMENTED** [review]
👍🏻

File: pkg/okteto/kubetoken_client.go (hunk 3 above)
Comment (jLopezbarb) [reply pkg/okteto/kubetoken_client.go:85]: Thanks for explaining it :)

**Review by jLopezbarb — APPROVED** [review]

**Pull request merged**